src/components/clusterConfiguration/ClusterSshKey.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { useFormikContext } from 'formik';
+import { FormGroup, Switch } from '@patternfly/react-core';
+import { TextAreaField } from '../ui';
+import { ClusterConfigurationValues } from '../../types/clusters';
+import { getFieldId } from '../ui/formik/utils';
+import HelperText from '../ui/formik/HelperText';
+import { ImageInfo } from '../../api/types';
+
+import './ClusterSshKey.css';
+
+const sshPublicKeyHelperText = (
+  <>
+    Provide a SSH public key to debug OpenShift nodes after installation. To generate a new key, use
+    the <em>ssh-keygen -o</em> command and paste the value of <em>~/.ssh/id_rsa.pub</em> here.
+  </>
+);
+
+const sshPublicKeyLabel = 'Host SSH Public Key for troubleshooting after installation';
+
+type ClusterSshKeyProps = {
+  imageSshPublicKey: ImageInfo['sshPublicKey'];
+};
+
+const ClusterSshKey: React.FC<ClusterSshKeyProps> = ({ imageSshPublicKey }) => {
+  const { setFieldValue, initialValues, values } = useFormikContext<ClusterConfigurationValues>();
+  const isCheckboxVisible = !!imageSshPublicKey;
+  const [isChecked, setChecked] = React.useState(
+    (isCheckboxVisible && initialValues.sshPublicKey === imageSshPublicKey) ||
+      !initialValues.sshPublicKey,
+  );
+
+  const onChange = (newVal: boolean) => {
+    setChecked(newVal);
+    if (newVal) {
+      setFieldValue('sshPublicKey', imageSshPublicKey);
+    } else {
+      setFieldValue('sshPublicKey', initialValues.sshPublicKey);
+    }
+  };
+
+  React.useEffect(
+    () => {
+      if (isChecked || !values.sshPublicKey) {
```

Comment:
I think this condition is wrong. The effect should happen only if the imageSshPublicKey is set and values.sshPublicKey is not.

Reply:
Currently the value.sshPublicKey is updated even if sshPublicKey is not set. Which makes the user prompt to save the form even when there are no changes to it because the effect sets the values.sshPublicKey to `''`.

Reply:
If the user re-configures the discovery image with a new key, what do we do? Should we switch the switch off and keep the old ssh key value (which is what is currently set in cluster) or should we keep the switch on and prompt the user to 'save and validate' to update the value in cluster configuration as well?

Reply:
I think the second option is right - the value should be updated and user should save it. I PR changed to this behaviour.